### DISCUSSION TITLE
Deleting the older daily-saved database files automatically

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2892] [Ideas]
The function to save the database everyday already exists.  You need to enable it.  
https://navid200.github.io/xDrip/docs/Backup/SaveDaily.html  

The problem with it is that after a while, the internal storage may fill up.  It helps if xDrip automatically deletes old database files.  For example, if xDrip only keeps the saved database files for the past 3 weeks.

We have a database corruption issue that we have not resolved yet:  https://github.com/NightscoutFoundation/xDrip/issues/1280
In the meantime, I would like to enable the daily save of the database by default and inform users to enable it.
But, I prefer not to have to tell users to manually delete the old files on a regular basis.  I prefer if xDrip does that itself.

What would be the best way for doing this?
Is there an existing code in xDrip, or another project, that deletes files from the internal storage?

**Comment by jamorham** [comment]
I think we could have a retention period for auto saved databases and delete ones older than this period. I don't really want to enable saving it by default. Wouldn't the cloud backup be better as its more modern and doesn't have the same issues when restoring?

**Reply by Navid200** [reply]
But, that would require us to create a daily backup feature for Google Drive.
The reason I wanted to use the old database backup method is that we already have an existing daily backup feature.

Can we retire the old database backup method then?

**Reply by jamorham** [reply]
Doesn't the cloud backup already have a automatic daily backup feature?

**Reply by Navid200** [reply]
I will have to test to see how it would behave and work for someone who does not have a Google account.

> I don't really want to enable saving it by default.  

Would you please tell me why?

**Reply by jamorham** [reply]
Cloud backup probably wont work without google account and play services and although there is an option for local filesystem storage, its not really intended to be used like that. One thing to note is that if you login to google drive on PC and use the 3 dot menu against the backup file you can usually access the last 100 days worth of backup snapshots individually if there was a particular point in time you were interested in.

The database backup to file system would require the permission to be requested in addition and I think it is legacy compared to the cloud backup which is designed to supersede it. In terms of historical data, there is nightscout and tidepool upload which is probably better for long term analytics.. so I just don't think it would be appropriate to enable local database backup by default when database corruptions are very rare in normal usage and its something users wouldn't be expecting and would leave behind sensitive information when the app was uninstalled. Never mind that there is still an outstanding bug with restoring them and that automatic deletion isn't implemented yet.
